src/plugins/data_views/common/data_views/data_views.ts

```diff
@@ -762,10 +763,16 @@ export class DataViewsService {
    * @param skipFetchFields if true, will not fetch fields
    * @returns DataView
    */
-  async create(spec: DataViewSpec, skipFetchFields = false): Promise<DataView> {
+  async create({ id, ...restOfSpec }: DataViewSpec, skipFetchFields = false): Promise<DataView> {
     const shortDotsEnable = await this.config.get(FORMATS_UI_SETTINGS.SHORT_DOTS_ENABLE);
     const metaFields = await this.config.get(META_FIELDS);
 
+    const spec = {
+      id: id ?? `local-${uuid.v4()}`,
+      isPersisted: id !== undefined,
```

Comment:
this won't work correctly as if we save the data view later it this flag would need to be updated. Maybe its better to expose a method on DataView  that checks if id starts with `local` ?

Reply:
you can also leave this out as i am adding it in the next pr in series: https://github.com/elastic/kibana/pull/134756

Reply:
we should also delete the `id` from the local dataview when `createSavedObject`

Reply:
IMO it would be better to save the saved object with the generated id. The save method could simply update the isPersisted property. I'm not sure if the `local-` prefix is necessary.

Reply:
saved object ids are different than the generated ids, i think we can't just use the generated one for the saved object ?
also:
- i think having a public flag is a bad idea, anyone can manipulate that and thus make it wrong (setting it to false on an object that is persisted ?)
- having a flag on a spec makes it public
- creating a dataview with id on the spec will mark it as persisted, which is not always the case. (with the local prefix this continues to work correctly)
- having logic in place that will make sure the property value is correct seems much more error prone and harder to maintain than having a method on the instance that "calculates" the value.

Reply:
I think there is a chance for conflict. I might be wrong of course but there are other reasons listed above why I think making this a flag on spec is a bad idea.

Reply:
could we remove isPersisted from spec and make it a private property and expose a public getter ? and make sure we handle it correctly internally (when saving to saved object) ?

Reply:
>I might be wrong, but there are other reasons listed above why I think making this a flag on spec is a bad idea.

I think we can separate the id generation from isPersisted.

UUIDs are designed to prevent collisions even when generated on different machines. Its their central idea.

data views that have been saved have a 'version' field. Perhaps we can use that to infer whether its been saved and `isPersisted` can be a getter, as you stated.